test/integration/targets/azure_rm_virtualmachine/tasks/virtualmachine.yml

```diff
@@ -67,6 +67,9 @@
         publisher: Canonical
         sku: 16.04-LTS
         version: latest
+      custom_data: |
+        #!/bin/sh
+        echo "custom_data was executed" > /custom_data.txt
   register: output
```

Comment:
Hi, Nice work and thanks, I was stuck looking for this. 
While a tad unrelated, could I suggest that this is also added to azure_rm_virtualmachine_scaleset as well. I've already forked and patched my own version but not wanting to duplicate a PR here. 

TIA

Reply:
@ajmckee if you'd like to ride on my pull request, perhaps you could give me a commit ID that I can cherry-pick which uses the exact same approach I'm using here?

Reply:
@cryptobanana I am sure mine is terrible as I needed it in a sticky situation and just hacked it together, but here is https://github.com/ramjam-io/ansible/commit/14b5dcf78e019b2951697c6803822b7e7685fbd1

Reply:
@ajmckee I created a branch based on your commit https://github.com/cryptobanana/ansible/commit/36386bc3155634099e955c832f9baa2185c92293 , but there's a little work required:

The integration test (test/integration/targets/azure_rm_virtualmachine_scaleset/tasks/main.yml) won't work exactly the same way because the "result" registered in the task `azure_rm_virtualmachine_scaleset:` won't provide an ip address that we can use the same way I'm running the test in (test/integration/targets/azure_rm_virtualmachine/tasks/virtualmachine.yml) Do you want to take a stab at tying up the loose ends?

Reply:
Sure if I get a moment over the holidays I will.

Initial Thinking... 
This would be an issue alright as generally VSS are deployed behind a application gateway, which takes forever and a day to deploy (OKay in fairness it is about 20-30 min). It may be viable to have the machine that is launched perform an action back (curl -X POST testhost or something). I'm pretty new to this, but will relish the investigation.. just don't hold breath over the holidays for a solution.

Reply:
I see that https://github.com/ansible/ansible/pull/33223 provides support for a load balancer to be defined with a scale set now. Which would allow the test to be executed in a coherent manner with the virtual machine test. 

There are multiple issues really with vss support with quite a few options missing. Still thinking aloud on this one.